cms/static/js/views/course_outline.js

```diff
@@ -34,8 +34,15 @@ function(
 
         renderTagCount: function() {
             const contentId = this.model.get('id');
-            const tagCountsByUnit = this.model.get('tag_counts_by_unit')
-            const tagsCount = tagCountsByUnit !== undefined ? tagCountsByUnit[contentId] : 0
+            console.log("contentId", contentId);
+            const tagCountsByBlock = this.model.get('tag_counts_by_block')
+            console.log("tagCountsByBlock:", tagCountsByBlock);
+            if (contentId === 'block-v1:SampleTaxonomyOrg1+STC1+2023_1+type@course+block@course') {
+                console.log("ignoring this");
+                return
+            }
+            const tagsCount = tagCountsByBlock !== undefined ? tagCountsByBlock[contentId] : 0
+            console.log('lookup count???', tagsCount);
             var countModel = new TagCountModel({
                 content_id: contentId,
```

Comment:
@ChrisChV I've implemented the tags count and manage tags functionality for the course, and it's working fine.

However when implementing it for sections/subsections I almost got it working, but I'm facing an issue and would appreciate some insights/help on it.

Since now I send back all the tags counts for all the course blocks (previously it was only for units) from the backend, and this code is recursively rendered, the tags counts are being overridden by the parent block. So for example, when adding tags the section tags count overrides all the subsections under it (however not the units tag counts for some reason). 

In fact, initially there is the main "course block" like the one in the debug code above, that overrides everything to 0 tags (it's always 0 since we don't apply tags to the "course block"), and since it is the top most it sets everything to 0, so when I remove/ignore it then correctly includes the counts for the sections (not subsections).

I apologize if my explanation is a bit confusing 😅 , but would appreciate it you could take a look and see if you notice anything off about it.

Reply:
I figured it out. Previously we had the following code:

```js
tagCountView = new TagCountView({el: this.$('.tag-count'), model: countModel});
```

which at each recursive level would override the child element since they all have the `tag-count` class. So in order to narrow down the query I changed it to 

```js
this.$(`.tag-count[data-locator="${contentId}"]`);
```

which made sure that the appropriate element is selected. That fixed the issue. It working as expected now.